vcluster/configure/vcluster-yaml/control-plane/components/coredns.mdx

```diff
@@ -9,43 +9,176 @@ description: Configure CoreDNS.
+
+### Deploy embedded CoreDNS
+
 <ProAdmonition/>
 
-## Separate CoreDNS
+The embedded CoreDNS feature allows you to run CoreDNS as part of the syncer, which saves the overhead of an external CoreDNS pod. This enterprise feature consolidates all virtual cluster components into a single pod, reducing pod count and optimizing resource allocation.
+
+The embedded approach requires consideration of operational trade-offs. Troubleshooting might become more complex since DNS and sync functions share the same container and process space. Updates and restarts affect both DNS and sync capabilities simultaneously, though all CoreDNS features remain supported.
+
+## CoreDNS uses port 1053
+
+CoreDNS in virtual clusters listens on port 1053 instead of the default DNS port 53 to avoid conflicts with the host cluster's DNS.
+
+The host cluster already operates its own DNS service on the standard port 53, so the virtual cluster DNS services cannot bind to the same port. Since multiple virtual clusters might run on the same host cluster, each requires its own DNS service that operates independently.
+
+Port 53 is classified as a privileged port on most operating systems and requires elevated permissions to bind to it. By using port 1053, CoreDNS containers can operate with standard user privileges.
+
+### Consider cloud provider networking
+
+Different cloud providers have varying network policies that can affect DNS traffic on port 1053.
+
+EKS clusters might require security group configuration to allow inbound traffic on port 1053 between worker nodes.
+
+GKE and AKS typically allow internal cluster traffic by default, but custom network policies may need to explicitly permit UDP traffic on port 1053 between cluster nodes.
+
+## Recognize DNS isolation and limitations
+
+Virtual cluster DNS isolation maintains clear boundaries between virtual clusters while creating specific challenges for cross-cluster communication.
+
+### DNS boundaries
+
+Virtual clusters cannot directly access host services via DNS, and host cluster pods cannot access virtual cluster services through the virtual cluster's DNS.
+
+When a pod inside a virtual cluster attempts to resolve a DNS name for a service that exists only in the host cluster, the virtual cluster CoreDNS instance returns an NXDOMAIN response because it maintains no knowledge of host cluster resources.
+
+### Apply service mapping strategies
+
+External name services provide an approach for mapping host cluster services into a virtual cluster's DNS namespace. This pattern creates a virtual cluster service that acts as a DNS alias.
```

Comment:
Why not use one of the existing features instead?
https://www.vcluster.com/docs/vcluster/configure/vcluster-yaml/networking/replicate-services
or 
https://deploy-preview-845--vcluster-docs-site.netlify.app/docs/vcluster/next/configure/vcluster-yaml/networking/resolve-dns